cpp/models/lct_secir/initializer_flows.h

```diff
@@ -74,145 +74,167 @@ class Initializer
      * The initial value vector is updated in the model.
      *
      * @param[in] total_population The total size of the considered population.
      * @param[in] deaths Number of deceased people from the disease at time 0.
      * @param[in] total_confirmed_cases Total number of confirmed cases at time 0.
-     * @return Returns true if one (or more) constraint(s) of the model or the initial flows are not satisfied, otherwise false. 
+     * @return Returns true if one (or more) constraint(s) of the model, the initial flows 
+     *      or the computed initial value vector are not satisfied, otherwise false. 
      */
-    bool compute_initialization_vector(ScalarType total_population, ScalarType deaths, ScalarType total_confirmed_cases)
+    bool compute_initialization_vector(Eigen::VectorXd const& total_population, Eigen::VectorXd const& deaths,
+                                       Eigen::VectorXd const& total_confirmed_cases)
     {
 
-        Eigen::VectorXd init(LctState::Count);
+        Eigen::VectorXd init(m_model.populations.get_compartments());
+
+        bool error = compute_initialization_vector_impl(init, total_population, deaths, total_confirmed_cases);
+        if (error) {
+            return true;
+        }
+
+        // Update initial value vector of the model.
+        for (size_t i = 0; i < m_model.populations.get_num_compartments(); i++) {
+            m_model.populations[i] = init[i];
+        }
+
+        // Check if constraints are fulfilled.
+        return check_constraints();
+    }
+
+    /**
+     * @brief Setter for the tolerance used to calculate the maximum support of ErlangDensity%s.
+     *
+     * @param[in] new_tol New tolerance.
+     */
+    void set_tol_for_support_max(ScalarType new_tol)
+    {
+        m_tol = new_tol;
+    }
+
+private:
+    /**
+     * @brief Implementation of the calcultion of the initial value vector slice that corresponds to a specified group.
+     *
+     * Computes a vector that can be used for the initalization of an LCT model stratified by groups with the number 
+     * of persons for each subcompartment. The groups are calculated recursively.
+     *
+     * @tparam Group The group for wich the corresponding slice of the initial value vector is calculated.
+     * @param[out] init The initial value vector under consideration.
+     * @param[in] total_population The total size of the considered population.
+     * @param[in] deaths Number of deceased people from the disease at time 0.
+     * @param[in] total_confirmed_cases Total number of confirmed cases at time 0.
+     * @return Returns true if one (or more) constraint(s) of the computed initial value vector are not satisfied,
+     *       otherwise false. 
+     */
+    template <size_t Group = 0, std::enable_if_t<(Group < Model::m_groups) && (Group >= 0), bool> = true>
+    bool compute_initialization_vector_impl(Eigen::VectorXd& init, Eigen::VectorXd const& total_population,
```

Comment:
Move this function before compute_initialization_vector?

Reply:
I actually want to keep this function as a private function. Therefore, i wouldn't switch the order.